pkg/operator/controller/controller_lb.go

```diff
@@ -68,12 +105,22 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 
 	service.Spec.Selector = IngressControllerDeploymentPodSelector(ci).MatchLabels
 
+	haveExternalLB := ci.Status.EndpointPublishingStrategy.LoadBalancer != nil && ci.Status.EndpointPublishingStrategy.LoadBalancer.Scope == operatorv1.ExternalLoadBalancer
+
+	if service.Annotations == nil {
+		service.Annotations = map[string]string{}
+	}
 	if infraConfig.Status.Platform == configv1.AWSPlatformType {
-		if service.Annotations == nil {
-			service.Annotations = map[string]string{}
-		}
 		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
 	}
+	if !haveExternalLB {
```

Comment:
Random style/readability thing — instead of "if not have external lb then add internal annotation" you could invert to "if internal lb then add internal annotation"

Reply:
I went back and forth on this because I am not sure it is always obvious that "internal" implies "not external".  I considered `internalOnly`, but that just moves the negative.  I'll change it to `isInternal` here.  Would you like the same change in `pkg/operator/controller/dns/controller.go` for consistency?

Reply:
Not a big deal, just something that stood out at me, no objection to keeping as-is